Intersect.Server/Entities/Entity.cs

```diff
@@ -1753,6 +1753,54 @@ public virtual void TryAttack(Entity target)
             );
             }
 
+            //Check on each attack if the enemy is a player AND if they are blocking.
+            if (enemy is Player player && player.Blocking)
+            {
+                //Getting the ID and Item the player is currently using.
+                var itemId = player.Items[player.Equipment[Options.ShieldIndex]].ItemId;
+                var item = ItemBase.Get(itemId);
+
+                if (item != null)
+                {
+                    var originalBaseDamage = baseDamage;
+                    var blockChance = item.BlockChance;
+                    var blockAmount = item.BlockAmount / 100.0;
+                    var blockAbsorption = item.BlockAbsorption / 100.0;
+
+                    //Generate a new attempt to block
+                    if (Randomization.Next(1, 101) < blockChance)
```

Comment:
what's the main goal of this randomization (?)
if you want to randomize from 1% to 100% then the logic is fine.
but if u want to rand from 0% to 100% (assuming that you are somewhat simulating percentage this way)
then it's 0,101